includes/specials/SpecialUserrights.php

```diff
@@ -524,17 +553,14 @@ private function groupCheckboxes( $usergroups, $user ) {
 			$set = in_array( $group, $usergroups );
 			# Should the checkbox be disabled?
 			$disabled = !(
-				( $set && $this->canRemove( $group ) ) ||
-				( !$set && $this->canAdd( $group ) ) );
+				( $set && self::canRemove( $user, $group, $this->isself ) ) ||
```

Comment:
the canAdd() and canRemove() eventually call PermissionsService::getChangeableGroups() every time they are executed and we do it in a loop that goes through all the existing groups, I can see some room for optimizing this flow